runtime/src/accounts_db.rs

```diff
@@ -1351,6 +1392,53 @@ impl AccountsDB {
         hasher.result()
     }
 
+    pub fn blake3_hash_account_data(
+        slot: Slot,
+        lamports: u64,
+        owner: &Pubkey,
+        executable: bool,
+        rent_epoch: Epoch,
+        data: &[u8],
+        pubkey: &Pubkey,
+    ) -> Hash {
+        if lamports == 0 {
+            return Hash::default();
+        }
+
+        let mut hasher = blake3::Hasher::new();
+        let mut buf = [0u8; 8];
+
+        LittleEndian::write_u64(&mut buf[..], lamports);
+        hasher.update(&buf);
+
+        LittleEndian::write_u64(&mut buf[..], slot);
+        hasher.update(&buf);
+
+        LittleEndian::write_u64(&mut buf[..], rent_epoch);
+        hasher.update(&buf);
```

Comment:
I know this is copypasta, but `u64` has a `to_le_bytes()` which is likely cheaper

Reply:
why cheaper?

Reply:
Is it not just a move?  It consumes the `u64`

Reply:
to_le_bytes() constructs a slice. naive compilation to me would be 1. expand stack for slice 2. write slice size/data where the write_u64 just does a single `mov` into buf which already exists on the stack.

Reply:
You got me thinking, so I went ahead and benched it. Seems to be ~3x faster

```
write_u64:   2.113420s
to_le_bytes: 0.611195s
```
for `it =  1_000_000` of
```
    for _ in 0..it {
        LittleEndian::write_u64(&mut buf[..], value);
        hasher.update(&buf);
    }
    ...
    for _ in 0..it {
        let v = value;
        hasher2.update(&v.to_le_bytes());
    }
```

Reply:
Wow, interesting. I would have thought it would be basically the same.

Reply:
Mind uploading the full source to a gist?

Reply:
Looks like I typoed the buffer length so the `write_u64` case was `update()`ing with too much data :facepalm: . After correcting, run times are, of course identical

Reply:
ah cool. That's what I expect.